src/main/java/com/pinterest/secor/util/orc/VectorColumnFiller.java

```diff
@@ -100,16 +103,31 @@ public void convert(JsonElement value, ColumnVector vect, int row) {
     }
 
     static class TimestampColumnConverter implements JsonConverter {
+        BackOffUtil back = new BackOffUtil(true);
+
         public void convert(JsonElement value, ColumnVector vect, int row) {
             if (value == null || value.isJsonNull()) {
                 vect.noNulls = false;
                 vect.isNull[row] = true;
             } else {
-                TimestampColumnVector vector = (TimestampColumnVector) vect;
-                vector.set(
-                        row,
-                        Timestamp.valueOf(value.getAsString().replaceAll(
-                                "[TZ]", " ")));
+                if (value.getAsJsonPrimitive().isString()) {
+                    TimestampColumnVector vector = (TimestampColumnVector) vect;
+                    vector.set(
+                            row,
+                            Timestamp.valueOf(value.getAsString().replaceAll(
+                                    "[TZ]", " ")));
+                } else if (value.getAsJsonPrimitive().isNumber()) {
+                    TimestampColumnVector vector = (TimestampColumnVector) vect;
+                    vector.set(
+                            row,
+                            new Timestamp(value.getAsLong()));
+                } else {
+                    if (!back.isBackOff()) {
+                        LOG.warn("Timestamp format is neither String nor Number");
```

Comment:
Do you want to show the error 'value' value (and with its type as well)?  It will help the debugging.